src/mask.js

```diff
@@ -110,9 +110,7 @@ angular.module('ui.mask', [])
                                             if (Object.prototype.hasOwnProperty.call(original, i)) {
                                                 if (current[i] === undefined) {
                                                     current[i] = angular.copy(original[i]);
-                                                } else {
```

Comment:
I'm not sure if simply removing the extend here is the best option? Can we guard against this specific case in another way?